Improve wording for 'ek' param (#99)
cose-wg/HPKE: draft-ietf-cose-hpke.md

that are specific to each COSE HPKE mode
so that they are fully specified, as required by {{RFC9864}}.

> In both cases, the unprotected header MUST contain a new COSE header parameter named 'ek'.

Author: I can't think of a reason, other than it doesn't need to be protected.

I copied from the previous text which said it must be unprotected.

Maybe Hannes has an opinion.

> In both cases, the unprotected header MUST contain a new COSE header parameter named 'ek'.

Reviewer: ```suggestion
In both cases, the new COSE header parameter 'ek' MUST be present.
```

At least when using Integrated Encryption, this can be a protected header parameter.

Author: Mike, I can go either way on this, but it may simplify implementation a little if you know it is always unprotected.

Interested in other opinions.

Author: I looked at my t_cose implementation and at other unprotected parameters like kid and content type. They all work with fine with either protected and unprotected, so I agree with you now, Mike.

This is a change in semantics from previous drafts, but I think it's what we should do.

I'll wait a while for other comments before accepting your change.

Reviewer 2: It is okay to _allow_ 'ek' to be protected (as this change does), but not okay to _require_ it (since it is nontrivial for sender). So yeah, seems fine.

And as said, the implementation I wrote can handle 'ek' in protected bucket just fine. Any allowed parameter can be protected if the algorithm allows protected parameters at all.
